aws/resource_aws_xray_encryption_config.go

```diff
@@ -0,0 +1,80 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/xray/waiter"
+)
+
+func resourceAwsXrayEncryptionConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsXrayEncryptionConfigPut,
+		Read:   resourceAwsXrayEncryptionConfigRead,
+		Update: resourceAwsXrayEncryptionConfigPut,
+		Delete: schema.Noop,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					xray.EncryptionTypeKms,
+					xray.EncryptionTypeNone,
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceAwsXrayEncryptionConfigPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).xrayconn
+
+	input := &xray.PutEncryptionConfigInput{
+		Type: aws.String(d.Get("type").(string)),
+	}
+
+	if v, ok := d.GetOk("key_id"); ok && v.(string) != "" {
```

Comment:
Nit: `!= ""` is already checked by `GetOk()`

```suggestion
	if v, ok := d.GetOk("key_id"); ok {
```